kornia/geometry/conversions.py

```diff
@@ -321,8 +321,21 @@ def rotation_matrix_to_angle_axis(rotation_matrix: torch.Tensor) -> torch.Tensor
 
     if not rotation_matrix.shape[-2:] == (3, 3):
         raise ValueError(f"Input size must be a (*, 3, 3) tensor. Got {rotation_matrix.shape}")
-    quaternion: torch.Tensor = rotation_matrix_to_quaternion(rotation_matrix, order=QuaternionCoeffOrder.WXYZ)
-    return quaternion_to_angle_axis(quaternion, order=QuaternionCoeffOrder.WXYZ)
+
+    axis = torch.zeros((rotation_matrix.shape[0], 3), device=rotation_matrix.device)
+    axis[:, 0] = rotation_matrix[:, 2, 1] - rotation_matrix[:, 1, 2]
+    axis[:, 1] = rotation_matrix[:, 0, 2] - rotation_matrix[:, 2, 0]
+    axis[:, 2] = rotation_matrix[:, 1, 0] - rotation_matrix[:, 0, 1]
+
+    # add epsilon for numerical stability
+    r = torch.norm(axis, dim=1).unsqueeze(1) + eps
+    t = rotation_matrix[:, 0, 0] + rotation_matrix[:, 1, 1] + rotation_matrix[:, 2, 2]
+    t = t.unsqueeze(1)
+    # use atan2 instead of torch.acos((t - 1)/2) for numerical stability
+    theta = torch.atan2(r, t - 1)
+    axis = axis / r
```

Comment:
```suggestion
    axis = axis / (r + eps)
```
Not sure about this one here

Reply:
:+1: